app/src/main/java/com/github/se/polyfit/model/post/Post.kt

```diff
@@ -98,6 +110,14 @@ data class Post(
       }
     }
 
+    private fun serializeLocalDate(data: LocalDate): Map<String, Any> {
```

Comment:
I actually think this and desrializeLocalDate could be extracted into a separate class as it can be used by other areas of the app

Reply:
LocalDate is a class directly provided by the Kotlin library. Locally, it can be directly used as a map due to its structure. However, when uploading `LocalDate` as is to Firebase, we encountered strange issues. Firebase tried to convert it into its own type, and when deserializing the post, the program crashed due to an unexpected type. In order to have symmetric serialization and deserialization, these classes were created to store it. Those functions should not be used in theory ouside of deserialization and serialization with the intent to store in on the cloud.

Reply:
No for some reason meal uses a complete different serialization method. Which I cannot remember why. It would be a great idea to utilize the same methods everywhere. But I think that it is outside the scope of this PR. What do you think ?

Reply:
I mean it is a 2 lines change.

Reply:
Yeah, can be done outside of this PR :D